src/agent.js

```diff
@@ -20,10 +20,17 @@ const onAuthorized = ( { socket, events } ) => ( agent ) => {
 		debug( 'received message', message )
 		events.emit( 'message', message )
 	} )
+
+	socket.on( 'system.info', done => {
+		events.emit( 'system.info', data => done( data ) )
+	} )
+
 	socket.on( 'role.add', ( role, done ) => {
 		debug( 'agent joining role', role )
+		events.emit( 'role.add', role )
```

Comment:
So we can selectively start the interval: https://github.com/Automattic/happychat/pull/121/files#diff-a42cba78a8f2c7c9988e1ba4165229bdR25

socket.io doesn't have any native events for room join/leave. Not strictly necessary as we could just have the interval running all the time but that seems like may be overkill if no one is listening for it.